src/openrct2/peep/Staff.cpp

```diff
@@ -1009,41 +1009,41 @@ bool Staff::DoMiscPathFinding()
  *
  *  rct2: 0x006C086D
  */
-static void staff_entertainer_update_nearby_peeps(Peep* peep)
+static void staff_entertainer_update_nearby_peeps(const Peep* entertainer)
 {
     for (auto guest : EntityList<Guest>(EntityListId::Peep))
     {
         if (guest->x == LOCATION_NULL)
             continue;
 
-        int16_t z_dist = abs(peep->z - guest->z);
+        int16_t z_dist = abs(entertainer->z - guest->z);
         if (z_dist > 48)
             continue;
 
-        int16_t x_dist = abs(peep->x - guest->x);
-        int16_t y_dist = abs(peep->y - guest->y);
+        int16_t x_dist = abs(entertainer->x - guest->x);
+        int16_t y_dist = abs(entertainer->y - guest->y);
 
         if (x_dist > 96)
             continue;
 
         if (y_dist > 96)
             continue;
 
-        if (peep->State == PEEP_STATE_WALKING)
+        if (guest->State == PEEP_STATE_WALKING)
         {
-            peep->HappinessTarget = std::min(peep->HappinessTarget + 4, PEEP_MAX_HAPPINESS);
+            guest->HappinessTarget = std::min(guest->HappinessTarget + 4, PEEP_MAX_HAPPINESS);
         }
-        else if (peep->State == PEEP_STATE_QUEUING)
+        else if (guest->State == PEEP_STATE_QUEUING)
         {
-            if (peep->TimeInQueue > 200)
+            if (guest->TimeInQueue > 200)
```

Comment:
This can be refactored to `guest->TimeInQueue = std::max(0, guest->TimeInQueue - 200);`

Reply:
Done. I got a little worried because `TimeInQueue` is unsigned, but I think the 'usual integer conversion' rules will apply here and it will be converted to signed int before subtraction/compare.